include/mbgl/style/sources/image_source.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <mbgl/style/source.hpp>
+#include <mbgl/util/image.hpp>
+#include <mbgl/util/optional.hpp>
+
+namespace mbgl {
+class LatLng;
+class AsyncRequest;
+
+namespace style {
+
+class ImageSource : public Source {
+public:
+    ImageSource(std::string id, const std::vector<LatLng>);
```

Comment:
Note that there would have to be separate types for mbgl and for the iOS/macOS SDK, because the iOS/macOS SDKs [can’t expose any C++ to the public](https://github.com/mapbox/mapbox-gl-native/blob/master/platform/ios/DEVELOPING.md#making-any-symbol-public).